src/averageAges.js

```diff
@@ -33,8 +46,29 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let filterWoman;
+
+  const hasChildren = (woman) => {
+    return people.some(person => person.mother === woman.name);
+  };
+
+  if (withChildren === undefined) {
+    filterWoman = people.filter(person => {
+      return person.sex === 'f';
+    });
+  } else {
+    filterWoman = people.filter(person => {
+      return person.sex === 'f';
+    }).filter(person => hasChildren(person));
```

Comment:
use single filter

Reply:
Try to rewrite without nested loops(As we discused on lesson), some have loop inside too